go-get-kubernetes.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# This script can be used while converting a repo from "dep" to "go mod"
+# by calling it after "go mod init" or to update the Kubernetes packages
+# in a repo that has already been converted. Only packages that are
+# part of kubernetes/kubernetes and thus part of a Kubernetes release
+# are modified. Other k8.io packages (like k8s.io/klog, k8s.io/utils)
+# need to be updated separately.
+
+set -o pipefail
+
+cmd=$0
+
+function help () {
+    echo "$cmd <kubernetes version = x.y.z> - update all components from kubernetes/kubernetes to that version"
+}
+
+if [ $# -ne 1 ]; then
+    help
+    exit 1
+fi
+case "$1" in -h|--help|help) help; exit 0;; esac
+
+die () {
+    echo >&2 "$@"
+    exit 1
+}
+
+k8s="$1"
+
+# If the repo imports k8s.io/kubernetes (directly or indirectly), then
```

Comment:
Do we actually have repos that do this? I checked csi-lib-utils and external-provisioner, and didn't find any kubernetes/kubernetes dependencies.

Reply:
`k8s.io/kubernetes` does get pulled in at the moment:
- https://github.com/kubernetes-csi/external-provisioner/tree/master/vendor/k8s.io/kubernetes
- https://github.com/kubernetes-csi/csi-driver-host-path/tree/master/vendor/k8s.io/kubernetes

For the driver it is the mount code which still hasn't been moved into a staging repo (there is work under way, I just don't remember how far it got). Not sure for `external-provisioner`, haven't checked.

Reply:
mount dependency is expected and will soon be removed. In general, we should start trying to cleanup and avoid any k8s.io/kubernetes dependencies.  But I guess it's fine to have this script in the meantime.

Does this help with the non-k8s.io/kubernetes repos (like k8s.io/api)?

Reply:
k8s.io/api is from k/k and gets updated by the script.

k8s.io/util isn't, but doesn't tag releases. k8s.io/klog has separate releases and shows up in `go.mod` with a proper version number (`k8s.io/klog v0.3.1`). I think "go get" picks the version specified by some other go.mod file for those modules, just as it does for other modules outside of Kubernetes.